src/test/java/hw2/ex2/Exercise2.java

```diff
@@ -0,0 +1,45 @@
+package hw2.ex2;
+
+import hw2.base.BaseTest;
+import org.testng.annotations.Test;
+
+public class Exercise2 extends BaseTest {
+
+    @Test
+    public void testingExercise_2() {
+        //1.Open test site by URL
+        openURL();
+
+        //2.Assert Browser title
+        assertTitle();
```

Comment:
**This comment for all methods you wrote:**
1. Let's imagine you have 2 pages. First page title is "Title 1". Second page title is "Title 2".
2. You have method `assertTitle();`. How can I use this method to assert title on each of two pages?

Feel problem? :)

It's not correct way to orginize your code.

In BaseTest you should have common methods (like login) or common assertions (which can be applied to different tests). Moreover this common method should be parameterized.

For example, you have:
```
    public void assertTitle() {
        assertEquals(driver.getTitle(), titleExpected);
    }
```
but should be:
```
    public void assertTitle(String expectedTitle) {
        assertEquals(driver.getTitle(), expectedTitle);
    }
```
and use it in test like:
 ```
  @Test
    public void test() {
        //1.Open test site by URL
        openURL();

        //2.Assert Browser title
        assertTitle("Title 1");
}

```

Reply:
**While refactor do not forget to extract repetitive actions to common methods**